### PR TITLE
Use pacote second attempt

### PR DESCRIPTION
**Pull request by jablko**
@andrewbranch Thanks for trying https://github.com/microsoft/DefinitelyTyped-tools/pull/451, I've now moved the cache into `os.tmpdir()`, to address https://github.com/microsoft/DefinitelyTyped-tools/pull/480#issue-1266598918. Would you be willing to give it another go, when you have a chance?

### CODE REVIEW

File: packages/utils/src/npm.ts [hunk 1]
```diff
@@ -1,157 +1,19 @@
...
-  const out: string[][] = [];
-  for (let i = 0; i < names.length; i += chunkSize) {
-    out.push(names.slice(i, i + chunkSize));
-  }
-  return out;
-}
+export const defaultCacheDir = joinPaths(
+  process.env.GITHUB_ACTIONS ? joinPaths(__dirname, "..") : os.tmpdir(),
+  "cache"
+);
```
Comment (jablko) [inline packages/utils/src/npm.ts:16]: Moved the cache into `os.tmpdir()`, to address https://github.com/microsoft/DefinitelyTyped-tools/pull/480#issue-1266598918.

Comment (andrewbranch) [reply packages/utils/src/npm.ts:16]: Verified that Node can write here on the Functions host over SSH, so I’m open to giving this another try. Using pacote seems like a worthwhile simplification if we can get it to work on our infrastructure. This week is busy for me though. I saw @sandersn merged a PR in this repo today so maybe he has some time?

Comment (sandersn) [reply packages/utils/src/npm.ts:16]: I'm on DT next week so I'll look then.

**Comment by andrewbranch** [comment]
yolo

**Comment by jablko** [comment]
@andrewbranch Thanks! :pray:

**Comment by andrewbranch** [comment]
I believe it didn’t crash, but it also didn’t have anything to publish.

image

**Comment by andrewbranch** [comment]
Actually, it looks like this is the old code, isn’t it? Because it looks like you removed the "Writing npm cache" log. The new version got published at 10:58 so I thought it would have used it here.

**Comment by jablko** [comment]
That's right, looks like the version before this PR. Maybe the next run will use the new version?

**Comment by jablko** [comment]
I take that as a sign it's using the new version, filling the cache.

**Comment by andrewbranch** [comment]
It’s still going! 😄

**Comment by jablko** [comment]
https://github.com/microsoft/DefinitelyTyped-tools/pull/453#issue-1227242953 speeds that up +/- 10 times.

**Comment by andrewbranch** [comment]
Oops, I should have just merged that one too

**Comment by andrewbranch** [comment]
Seemed to work... will watch the next run to make sure it successfully uses the new cache it built

**Comment by andrewbranch** [comment]
Yep. I think this is good. Thanks! 🎉

**Comment by jablko** [comment]
Thanks a lot!

**Pull request merged**